datafusion/core/tests/sql/parquet_schema.rs

```diff
@@ -141,6 +139,13 @@ async fn schema_merge_can_preserve_metadata() {
     let table_path = table_dir.to_str().unwrap().to_string();
 
     let ctx = SessionContext::new();
+
+    // explicitly disable schema clearing
+    ctx.config_options()
```

Comment:
Ideally we will have a nicer API for this: https://github.com/apache/arrow-datafusion/issues/3908